### PR TITLE
Optional positionable attributes and strand

### PR DESCRIPTION
**Pull request by xgaia**
- [x] Positionable attributes are now optional (except for start and end)
- [x] We can filters on common positionable attributes between the 2 entities (same ref/tax)
- [x] We can filter on same/opposite/both strand (if strand attributes is define in the 2 entities)

selection_046

- [x] Indent the downloaded json (download query)
- [x] launch query with enter key
- [x] Update `startAskomics.sh` with --prod argument for gulp
- [x] Update readme.md

### CODE REVIEW

File: askomics/ask_view.py [hunk 1]
```diff
@@ -180,6 +180,48 @@ def get_list_named_graphs(self):
 
         return namedGraphs
 
+    @view_config(route_name='positionable_attr', request_method='POST')
+    def positionable_attr(self):
```
Comment (ofilangi) [inline askomics/ask_view.py]: Add a test for this function

File: askomics/libaskomics/rdfdb/SparqlQueryBuilder.py [hunk 2]
```diff
@@ -114,3 +114,21 @@ def get_metadatas(self, graph):
             WHERE {	GRAPH <"""+self.get_param("askomics.graph")+""">
 		        { <""" + graph + """> ?p ?o
                 VALUES ?p {prov:generatedAtTime dc:creator dc:hasVersion prov:describesService prov:wasDerivedFrom} } }""")
+
+    def get_if_positionable(self, uri):
```
Comment (ofilangi) [inline askomics/libaskomics/rdfdb/SparqlQueryBuilder.py]: Add a test for this function

File: askomics/libaskomics/rdfdb/SparqlQueryBuilder.py [hunk 3]
```diff
@@ -114,3 +114,21 @@ def get_metadatas(self, graph):
...
+        return self.prepare_query(
+        """SELECT DISTINCT ?exist
+        WHERE {
+            GRAPH <"""+self.get_param("askomics.graph")+"""> { ?g rdfg:subGraphOf <"""+self.get_param("askomics.graph")+""">}
+            BIND(EXISTS {<"""+uri+"""> displaySetting:is_positionable "true"^^xsd:boolean} AS ?exist)
+        }""")
+
+    def get_common_positionable_attributes(self, uri1, uri2):
```
Comment (ofilangi) [inline askomics/libaskomics/rdfdb/SparqlQueryBuilder.py]: add a test for this function

File: askomics/static/js/AskomicsForceLayoutManager.js [hunk 4]
```diff
@@ -163,6 +163,13 @@ var AskomicsForceLayoutManager = function () {
       ctrlPressed = false ;
   });
 
+  // Trigger query with enter key
```
Comment (ofilangi) [inline askomics/static/js/AskomicsForceLayoutManager.js]: why ctrlPressed variable is remove ? the multiple selection still works ?

File: askomics/static/js/link/AskomicsPositionableLink.js [hunk 5]
```diff
@@ -7,36 +7,80 @@ class AskomicsPositionableLink extends AskomicsLink {
...
+    this.position_strand = obj.position_strand ;
   }
   getPanelView() {
     return new AskomicsPositionableLinkView(this);
   }
 
-  getFillColor() { return 'darkgreen'; }
+  getTextFillColor() { return 'darkgreen'; }
 
   buildConstraintsSPARQL() {
```
Comment (ofilangi) [inline askomics/static/js/link/AskomicsPositionableLink.js]: a test which pass inside new test.

**Comment by ofilangi** [comment]
Hello xgaia, Thx for this add. I comment somme line to improv this dev.

**Comment by xgaia** [comment]
Hi, Ok I will look at this after my "soutenance" :-)

**Comment by xgaia** [comment]
- [x] Test for `pos_attributes` function
- [x] fix a little bug when user keep the enter key pressed
- [x] add css and files for font awsome and google materials icons
+ google cloud icon: `<i class="material-icons md-18">cloud</i>`
+  fa cloud icon: `<i class="fa fa-cloud"></i>`

**Pull request merged**